cypress/integration/admin/login/login.spec.js

```diff
@@ -17,4 +19,146 @@ context("Login Page", () => {
             cy.findByText(Cypress.env("DEFAULT_ADMIN_USER_USERNAME")).should("exist");
         });
     });
+
+    it("try to access admin pages as anonymous user", () => {
+        cy.visit("/page-builder/pages");
+        cy.findByText(/sign in/i).should("exist");
+        cy.findByText(/forgot password?/i).should("exist");
+
+        cy.visit("/form-builder/forms");
+        cy.findByText(/sign in/i).should("exist");
+        cy.findByText(/forgot password?/i).should("exist");
+
+        cy.visit("/cms/content-models");
+        cy.findByText(/sign in/i).should("exist");
+        cy.findByText(/forgot password?/i).should("exist");
+    });
+
+    it("try to access non-existing page as logged user", () => {
+        cy.visit("/");
+        cy.findByText(/sign in/i).should("exist");
+        cy.findByText(/forgot password?/i).should("exist");
+
+        cy.findByLabelText(/your e-mail/i).type("admin@webiny.com");
+        cy.findByLabelText(/your password/i).type("12345678");
+        cy.findByText(/submit/i).click();
+
+        cy.findByPlaceholderText(/search.../i).should("exist");
+        cy.findByText(/pages/i).should("exist");
+
+        cy.visit("/page-builder/pagessss");
+        cy.findByText(/The route is either missing, or you're not authorized to view it./i).should(
+            "exist"
+        );
+        cy.findByText(/Please contact your administrator to report the issue./i).should("exist");
+        cy.findByText(/Take me back./i).should("exist");
+    });
+
+    it("authenticated user must logout successfully", () => {
+        cy.visit("/");
+        cy.findByText(/sign in/i).should("exist");
+        cy.findByText(/forgot password?/i).should("exist");
+
+        cy.findByLabelText(/your e-mail/i).type("admin@webiny.com");
+        cy.findByLabelText(/your password/i).type("12345678");
+        cy.findByText(/submit/i).click();
+
+        cy.findByPlaceholderText(/search.../i).should("exist");
+        cy.findByText(/pages/i).should("exist");
+
+        cy.findByTestId("logged-in-user-menu-avatar").click();
+        cy.findByText(/Sign out/i).click();
+        cy.findByText(/sign in/i).should("exist");
+        cy.findByText(/forgot password?/i).should("exist");
+    });
+
+    it('recover password E2E for "full-access" user', () => {
+        const serverId = 'z1fihlo8';
+        let passwordResetCode;
+        const newUserAccountEmail = uniqid("cypress_", "@z1fihlo8.mailosaur.net");
+
+        let fullAccessGroupUser;
```

Comment:
Just a heads up here - I've just approved test execution in our CI/CD, and these couple of lines will be reported by prettier / eslint. Make sure to check the run logs.

https://github.com/webiny/webiny-js/actions/runs/2207349371